tests/test_user_profile.py

```diff
@@ -4,90 +4,56 @@
+        assert new_name == profile_details.user_profile_name
+        assert new_user['url'] == profile_details.user_profile_url
+        assert new_user['email'] == profile_details.bugzilla_email
+
+        edit_profile = profile_details.click_edit_profile_button()
+        assert new_name == edit_profile.display_name
+        assert new_username == edit_profile.username
+        assert new_user['url'] == edit_profile.user_profile_url
+        assert new_user['email'] == edit_profile.bugzilla_email
+
+        confirm_delete = edit_profile.click_delete_profile_button()
+        assert confirm_delete.is_the_current_page
+
+        homepage = confirm_delete.click_cancel_button()
+        assert homepage.is_the_current_page
+        assert homepage.is_user_logged_in
+        assert new_name.upper() in homepage.header.profile_link_text
+        assert new_name in homepage.displayed_profile_name
+
+        profile_details = home_page.header.click_user_profile_details()
+        edit_profile = profile_details.click_edit_profile_button()
+        confirm_delete = edit_profile.click_delete_profile_button()
+        home_page = confirm_delete.click_confirm_button()
         assert not home_page.is_user_logged_in
 
-        logged_in_home_page = home_page.login(existing_user, 'home_page')
-        assert logged_in_home_page.is_the_current_page
-        assert logged_in_home_page.is_user_logged_in
-
-        # click user profile details link
-        user_profile_details_page = logged_in_home_page.header.click_user_profile_details()
-        assert user_profile_details_page.is_the_current_page
-
-        # click edit profile button
-        user_profile_edit_page = user_profile_details_page.click_edit_profile_button()
-        assert user_profile_edit_page.is_the_current_page
-        assert existing_user['profile']['name'] == user_profile_edit_page.display_name
-        assert existing_user['profile']['username'] == user_profile_edit_page.username
-
-        # change display name and username
-        new_display_name = re.sub(r'[\W_]+', '', existing_user['profile']['name'] + str(time.time()))
-        new_username = re.sub(r'[\W_]+', '', existing_user['profile']['username'] + str(time.time()))
-        user_profile_edit_page.enter_name(new_display_name)
-        user_profile_edit_page.enter_username(new_username)
-        user_profile_details_page = user_profile_edit_page.click_save_button('user_profile_details')
-
-        assert user_profile_details_page.is_the_current_page
-        assert new_display_name == user_profile_details_page.user_profile_name
-
-        # edit profile again
-        user_profile_edit_page = user_profile_details_page.click_edit_profile_button()
-        assert new_display_name == user_profile_edit_page.display_name
-        assert new_username == user_profile_edit_page.username
+        create_profile = home_page.login(new_user)
+        assert create_profile.is_user_logged_in
+        assert create_profile.is_the_current_page
```

Comment:
No good reason. I was just "cleaning up" and figured I'd combine them into one test, but really there should be two - one for editing and one for deleting, so I will split them again.